src/Plugin/Hal.php

```diff
@@ -114,6 +114,14 @@ class Hal extends AbstractHelper implements
      */
     protected $urlHelper;
 
+
+    /**
+     * Maximum number of nesting levels
+     *
+     * @var integer
+     */
+    protected $maxDepth;
```

Comment:
Could this inhibit a developer's understanding of what it can do?  I would prefer a -1 ∞